tools/commands-cordova.js

```diff
@@ -638,38 +673,64 @@ var ensureCordovaPlugins = function (projectContext, options) {
     }
   });
 
-  if (shouldReinstallPlugins) {
+  if (! _.isEmpty(pluginsFromLocalPath))
+    verboseLog('Reinstalling cordova plugins added from the local path');
+
+  if (shouldReinstallPlugins || ! _.isEmpty(pluginsFromLocalPath)) {
     // Loop through all of the current plugins and remove them one by one until
-    // we have no plugins. It's necessary to loop because we might have
-    // dependencies between plugins.
-    var uninstallAllPlugins = function () {
+    // we have deleted proper amount of plugins. It's necessary to loop because
```

Comment:
Basically the `uninstallPlugins` function now gets the `pluginsToUninstall` array. Now it can delete any amount of plugins so it is not allways all of them. By "proper" I meant that the amount of deleted plugins is equal to size of the array :) Anyway we could just say 'until we have deleted all of the plugins provided in pluginsToUninstall array' or sth like that. The two scenarios (delete all or just the plugins from the local path) are described in the comment on 639, maybe it should also be reminded before the conditional on 711.